tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyLog.java

```diff
@@ -82,66 +69,23 @@ public static void setDir(File homeFile) {
-            .append(LogField.DATE).append(' ').append(nextTimePost).append(' ').append(new Date()).append('\n')
-            .append(FIELD_SEP).append(LogField.REVISION).append(' ').append(SurveyMain.getCurrevCldrApps()).append(' ').append(SurveyMain.getNewVersion())
-            .append(' ').append(CLDRConfig.getInstance().getPhase()).append(' ').append(CLDRConfig.getInstance().getEnvironment()).append('\n')
-            .append(FIELD_SEP).append(LogField.UPTIME).append(' ').append(SurveyMain.uptime).append('\n');
-        if (ctx != null) {
-            sb.append(FIELD_SEP).append(LogField.CTX).append(' ').append(ctx).append('\n');
-        }
-        sb.append(FIELD_SEP).append(LogField.LOGSITE).append(' ').append(StackTracker.currentStack()).append('\n');
-        Throwable t = exception;
-        while (t != null) {
-            sb.append(FIELD_SEP).append(LogField.MESSAGE).append(' ').append(t.toString())
-                .append('\n');
-            sb.append(FIELD_SEP).append(LogField.STACK).append(' ').append(StackTracker.stackToString(t.getStackTrace(), 0))
-                .append('\n');
-            if (t instanceof SurveyException) {
-                sb.append(FIELD_SEP).append(LogField.SURVEYEXCEPTION).append(' ').append(((SurveyException) t).getErrCode())
-                    .append('\n');
-            }
-            if (t instanceof SQLException) {
-                SQLException se = ((SQLException) t);
-                sb.append(FIELD_SEP).append(LogField.SQL).append(' ').append('#').append(se.getErrorCode()).append(' ')
-                    .append(se.getSQLState()).append('\n');
-                t = se.getNextException();
-            } else {
-                t = t.getCause();
-            }
-        }
-
-        // First, log to file
-        // TODO move into chunkyreader
-        File baseDir = gBaseDir;
-        if (baseDir == null || !baseDir.isDirectory()) {
-            setDir(new File("."));
-            System.err.println(SurveyLog.class.getName() + " Warning: Storing exception.log in " + gBaseDir.getAbsolutePath()
-                + "/exception.log");
-        }
-        try {
-            File logFile = new File(baseDir, "exception.log");
-            OutputStream file = new FileOutputStream(logFile, true); // Append
-            PrintWriter pw = new PrintWriter(new OutputStreamWriter(file, "UTF-8"));
-            pw.append(sb);
-            pw.close();
-            file.close();
-        } catch (IOException ioe) {
-            logger.severe("Err: " + ioe + " trying to log exceptions!");
-            ioe.printStackTrace();
-        }
-
-        getChunkyReader().nudge();
-
-        // then, to screen
-        logger.severe(sb.toString());
+    public static void logException(Logger logger, final Throwable exception, String what, WebContext ctx) {
+        // TODO: needs POM entry
+        // LogRecordContext.addExtension("webContext",ctx.toString());
```

Comment:
Is this TODO tied to completion of the ticket?

Reply:
yes